tide_site_restriction.module

```diff
@@ -162,3 +158,81 @@ function tide_site_restriction_form_views_exposed_form_alter(&$form, FormStateIn
     }
   }
 }
+
+/**
+ * Implements hook_node_access_records().
+ */
+function tide_site_restriction_node_access_records(Node $node) {
+  $grants = [];
+  if (!$node->isPublished()) {
+    $sites_values = $node->get('field_node_site')->getValue();
+    if (empty($sites_values)) {
+      return $grants;
+    }
+    $sites = array_column($sites_values, 'target_id');
+    foreach ($sites as $site) {
+      $grants[] = [
+        'realm' => 'tide_site_restriction',
+        'gid' => $site,
+        'grant_view' => 1,
+        'grant_update' => 0,
+        'grant_delete' => 0,
+        'nid' => $node->id(),
+      ];
+    }
+  }
+  return $grants;
+}
+
+/**
+ * Implements hook_node_grants().
+ */
+function tide_site_restriction_node_grants(AccountInterface $account, $op) {
+  $user = User::load($account->id());
+  $user_sites = $user->get('field_user_site')->getValue();
+  if (empty($user_sites)) {
+    return [];
+  }
+  $sites = array_column($user_sites, 'target_id');
+  $grants = [];
+  if ($op == 'view') {
+    foreach ($sites as $site) {
+      $grants['tide_site_restriction'][] = $site;
+    }
+    return $grants;
+  }
+}
+
+/**
+ * Implements hook_node_grants_alter().
+ */
+function tide_site_restriction_node_grants_alter(&$grants, AccountInterface $account, $op) {
+  foreach ($grants as $realm => $gid) {
+    // Other modules should respect to the tide_site_restriction module.
+    if ($realm != 'tide_site_restriction') {
+      unset($grants[$realm]);
+    }
+  }
+  return $grants;
+}
+
+/**
+ * Computes access result based on the sites assigned.
+ */
+function _compute_access(AccountInterface $account, EntityInterface $entity, Helper $helper) {
+  $user = User::load($account->id());
+  $user_sites = $helper->getUserSites($user);
+  $user_can_bypass_restriction = $helper->canBypassRestriction($account);
+  if (!$user_can_bypass_restriction) {
```

Comment:
Do you have to do an `AccessResult::allowed()` here as the else part of this statement?

Reply:
Hi @anthony-malkoun ,
on line 55 https://github.com/dpc-sdp/tide_site_restriction/blob/feature/SDPA-3805-only-view-draft-approver-and-editor/tide_site_restriction.module#L55
it will return `return AccessResult::neutral()->addCacheableDependency($entity);`